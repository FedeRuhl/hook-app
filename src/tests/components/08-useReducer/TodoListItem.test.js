import { shallow } from 'enzyme';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';
import demoTodos from "../../fixtures/demoTodos";

describe('TodoListItem tests', () => {

    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(
    <TodoListItem
        todo={demoTodos[0]}
        i={0}
        handleToggle={ handleToggle}
        handleDelete={ handleDelete }
    />
    );

    test('should return TodoListItem successfull', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should call delete function', () => {
        
        wrapper.find('button').first().simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('should call toggle function', () => {
        wrapper.find('p').first().simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    });
    
    test('should show text paragraph correctly', () => {
        const p = wrapper.find('p').first().text();
        expect(p).toBe(`1. ${demoTodos[0].desc}`);
    });

    test('should has complete class if done is true', () => {
        const todo={
            'id': 1,
            'desc': 'learn to play the piano',
            'done': true
        };

        const wrapper = shallow(
            <TodoListItem
                todo={todo}
                i={0}
                handleToggle={ handleToggle}
                handleDelete={ handleDelete }
            />
        );

        expect(wrapper.find('p').hasClass('complete')).toBe(true);
    });
    
});