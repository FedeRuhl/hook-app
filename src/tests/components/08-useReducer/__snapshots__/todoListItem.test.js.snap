// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem tests should return TodoListItem successfull 1`] = `
<li
  className="list-group-item"
>
  <p
    className=""
    onClick={[Function]}
  >
    1
    . 
    learn react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
